### DISCUSSION TITLE
Search in Custom Libraries?

### DISCUSSION THREAD

**Discussion by Michael-B-Morell-CISSP** [jgraph/drawio#4546] [Q&A]
Community, does anyone know of a secret sauce that allows you to search in custom shape libraries?

When I add a custom shape library and it is populated with any shape (custom built or from the web), and I try to use the search function, it will never show up in the results.

I am sure that I am missing something here; have tried it on desktop app, drawio for confluence, and on my own docker drawio server.  All 3 to no avail.

Thanks in advance.

**Comment by davidjgraph** [comment]
No, that feature doesn't currently exist.

**Comment by Michael-B-Morell-CISSP** [comment]
This email contains sensitive information

Thank you David for confirming that I am not insane (-;

Out of morbid curiosity, is it on the dev roadmap?  It is kind of interesting that it was not baked in already; especially for those of us who have huge amounts of custom libraries.

With Warm Regards,

Michael B. Morell, CISSP #431307 CCSK (he/him/his/sir)
DirectionWeb

**Reply by Oattsplease** [reply]
it doesn't exist in the web app, but if you're creating libraries programmatically or editing the text manually, at the end of the JSON you can add a 'title' field with any text you'd like to search the shape by, and then it will show up in the search field

```
/mxGraphModel&gt;",
    "w": 330.002222222222,
    "h": 256,
    "aspect": "fixed",
    "title":"enter your text here!!!"
  }
```

**Comment by Michael-B-Morell-CISSP** [comment]
This email contains sensitive information

Sweet! Thank you Matt!!!  That worked like a charm.


With Warm Regards,



Michael B. Morell, CISSP #431307 CCSK (he/him/his/sir)

DirectionWeb

Owner and Information Security Professional
***@***.***
C: (954) 802-5736

O: (754) 400-7741



________________________________
